auth/auth.go

```diff
@@ -160,6 +161,53 @@ func (auth *Authenticator) GetRoleIncDeleted(name string) (Role, error) {
 	return role, err
 }
 
+func (auth *Authenticator) InheritedChannels(user User, princ Principal) error {
+	cumulativeChannels := ch.TimedSet{}
```

Comment:
We can probably save a bit of memory by counting as we're iterating, rather than storing what channels we see then grabbing length.